### DISCUSSION TITLE
Relationship SAWarning

### DISCUSSION THREAD

**Discussion by IvesFeng** [sqlalchemy/sqlalchemy#8146] [Usage Questions] [answered]
I am using automap to automatically generate classes.
```
class deployment_info(Base):
    __tablename__ = 'deployment_info'

    software_id = Column(ForeignKey("software_info.id"))
    softwares = relationship("software_info", back_populates='deployments')


class software_info(Base):
    __tablename__ = 'software_info'

    deployments = relationship('deployment_info', back_populates="softwares")
```
There will be a warning
`__main__:1: SAWarning: relationship 'software_info.**deployment_info_collection**' will copy column software_info.id to column deployment_info.software_id, which conflicts with relationship(s): 'deployment_info.softwares' (copies software_info.id to deployment_info.software_id). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   To silence this warning, add the parameter 'overlaps="softwares"' to the 'software_info.deployment_info_collection' relationship. (Background on this error at: https://sqlalche.me/e/14/qzyx)`
How to avoid this warning?

**Comment by zzzeek** [comment, marked as answer]
automap is already creating relationships which are conflicting with the ones you are making. 

if you want to make your own relationships, dont use automap.  use the pattern at https://docs.sqlalchemy.org/en/14/orm/declarative_tables.html#mapping-declaratively-with-reflected-tables using imperative table or DeferredReflection.
